### PR TITLE
Enforce stricter transaction limits

### PR DESCRIPTION
**Pull request by graetzer**
### Scope & Purpose

Add transaction limitations for stream transactions

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new **integration tests** (i.e. in shell_server / shell_server_aql)

http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5748/

### CODE REVIEW

File: arangod/Transaction/Manager.h [hunk 1]
```diff
@@ -53,11 +53,14 @@ namespace transaction {
...
-/// @bried Tracks TransasctionState instances
+/// @brief Tracks TransasctionState instances
 class Manager final {
-  static constexpr size_t numBuckets = 16;
-  static constexpr double defaultTTL = 10.0 * 60.0;   // 10 minutes
-  static constexpr double tombstoneTTL = 5.0 * 60.0;  // 5 minutes
+  static constexpr size_t numBuckets   = 16;
+  static constexpr double idleTTL   = 10.0;              // 10 seconds
+  static constexpr double totalTTL  = 60.0;              // 60 seconds
+  static constexpr double totalTTLDBServer  = 3 * 60.0;  // 6 minutes
```
Comment (jsteemann) [inline arangod/Transaction/Manager.h]: the code and the comment disagree here. If it's 6 minutes, then the code should be `6 * 60.0`. If 3 minutes, then the comment should be adjusted.

File: arangod/Transaction/Manager.h [hunk 2]
```diff
@@ -47,52 +47,55 @@ struct TransactionData {
...
 
-/// @bried Tracks TransasctionState instances
+/// @brief Tracks TransasctionState instances
 class Manager final {
   static constexpr size_t numBuckets = 16;
-  static constexpr double defaultTTL = 10.0 * 60.0;   // 10 minutes
-  static constexpr double tombstoneTTL = 5.0 * 60.0;  // 5 minutes
-  
+  static constexpr double idleTTL = 10.0;                          // 10 seconds
+  static constexpr double idleTTLDBServer = 3 * 60.0;              // 6 minutes
```
Comment (jsteemann) [inline arangod/Transaction/Manager.h]: comment or code still wrong.

File: arangod/Transaction/Manager.cpp [hunk 3]
```diff
@@ -422,38 +449,37 @@ std::shared_ptr<transaction::Context> Manager::leaseManagedTrx(TRI_voc_tid_t tid
 
     ManagedTrx& mtrx = it->second;
     if (mtrx.type == MetaType::Tombstone) {
-      return nullptr; // already committet this trx
+      return nullptr;  // already committet this trx
```
Comment (jsteemann) [inline arangod/Transaction/Manager.cpp]:
```suggestion
      return nullptr;  // already committed this trx
```

File: arangod/Transaction/Manager.cpp [hunk 4]
```diff
@@ -45,19 +45,20 @@
 #include <velocypack/Iterator.h>
 #include <velocypack/velocypack-aliases.h>
 
-
 namespace arangodb {
 namespace transaction {
 
+const size_t Manager::maxTransactionSize;  // 128 MiB
```
Comment (jsteemann) [inline arangod/Transaction/Manager.cpp]: This `maxTransactionSize` limit is actually still enforced.
I understood from today's discussion with Ingo that we only want to have the idle timeout, but no other restrictions yet. Did I misunderstand? @ifcologne ?

Comment (graetzer) [reply arangod/Transaction/Manager.cpp]: I think the max size is still good, I thought its about the overall max time which we want to avoid

Comment (ifcologne) [reply arangod/Transaction/Manager.cpp]: We want to enforce `maxTransactionSize` (128MB) and `idleTTL` (10Sec).

We do not want to enforce the limit on the total lifespan of a transaction.

Comment (jsteemann) [reply arangod/Transaction/Manager.cpp]: Got it. I misunderstood it then. All good.

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5767/

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5774/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5775/

**Pull request merged**